tests/components/panasonic_viera/test_init.py

```diff
@@ -0,0 +1,123 @@
+"""Test the Panasonic Viera setup process."""
+from unittest.mock import Mock
+
+from asynctest import patch
+
+from homeassistant.components.panasonic_viera.const import (
+    CONF_APP_ID,
+    CONF_ENCRYPTION_KEY,
+    CONF_ON_ACTION,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DOMAIN,
+)
+from homeassistant.config_entries import ENTRY_STATE_NOT_LOADED
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+MOCK_CONFIG_DATA = {
+    CONF_HOST: "0.0.0.0",
+    CONF_NAME: DEFAULT_NAME,
+    CONF_PORT: DEFAULT_PORT,
+    CONF_ON_ACTION: None,
+}
+
+MOCK_ENCRYPTION_DATA = {
+    CONF_APP_ID: "mock-app-id",
+    CONF_ENCRYPTION_KEY: "mock-encryption-key",
+}
+
+
+def get_mock_remote():
+    """Return a mock remote."""
+    mock_remote = Mock()
+
+    async def async_create_remote_control(during_setup=False):
+        return
+
+    mock_remote.async_create_remote_control = async_create_remote_control
+
+    return mock_remote
+
+
+async def test_setup_entry_encrypted(hass):
+    """Test setup with encrypted config entry."""
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=MOCK_CONFIG_DATA[CONF_HOST],
+        data={**MOCK_CONFIG_DATA, **MOCK_ENCRYPTION_DATA},
+    )
+
+    mock_entry.add_to_hass(hass)
+
+    mock_remote = get_mock_remote()
+
+    with patch(
+        "homeassistant.components.panasonic_viera.Remote", return_value=mock_remote,
+    ):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+        state = hass.states.get("media_player.panasonic_viera_tv")
+
+        assert state
+        assert state.name == DEFAULT_NAME
+
+
+async def test_setup_entry_unencrypted(hass):
+    """Test setup with unencrypted config entry."""
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN, unique_id=MOCK_CONFIG_DATA[CONF_HOST], data=MOCK_CONFIG_DATA,
+    )
+
+    mock_entry.add_to_hass(hass)
+
+    mock_remote = get_mock_remote()
+
+    with patch(
+        "homeassistant.components.panasonic_viera.Remote", return_value=mock_remote,
+    ):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+        state = hass.states.get("media_player.panasonic_viera_tv")
+
+        assert state
+        assert state.name == DEFAULT_NAME
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5451397)